### PR TITLE
adding collatz-conjecture

### PR DESCRIPTION
**Pull request by Average-user**
It was on the Unimplemented exercises of [exercism.io](http://exercism.io) so i decided to implement it.

### CODE REVIEW

File: exercises/collatz-conjecture/project.clj [hunk 1]
```diff
@@ -0,0 +1,5 @@
+(defproject collatz-conjecture "0.1.0-SNAPSHOT"
+  :description "collatz-conjecture exercise"
+  ;;:url "https://github.com/exercism/xclojure/tree/master/exercises/collatz-conjecture"
+  :dependencies [[org.clojure/clojure "1.8.0"]
+                 [proto-repl "0.3.1"]])
```
Comment (haus) [inline exercises/collatz-conjecture/project.clj]: The `proto-repl` dependency should get pulled out I think.

**Comment by haus** [comment]
@Average-user thanks for much for the contribution. This looks like a great start.

File: exercises/collatz-conjecture/src/example.clj [hunk 2]
```diff
@@ -0,0 +1,8 @@
+(ns collatz-conjecture)
+
+(defn collatz [n]
+  (when (> n 0)
+    (loop [t 0 n n]
```
Comment (haus) [inline exercises/collatz-conjecture/src/example.clj]: Style nitpick: I would generally put each pair of bindings on their own line to disambiguate them.

Comment (Average-user) [reply exercises/collatz-conjecture/src/example.clj]: i would if there were more than 2, but if you insist i will.

**Comment by Average-user** [comment]
I did like this for  the Haskell repository and there were no problems, not yet at least.

But again ... If you insist , i can change it.

**Comment by Average-user** [comment]
Do you know any other exercise that handle with errors ? i'm not completely sure of how would i do it, if i have to.

**Comment by haus** [comment]
It looks like https://github.com/exercism/clojure/blob/4562cf14857a2b886e27344c2e5e56e20e7f99bc/exercises/wordy/src/example.clj#L16-L23 throws an exception, and the tests validate that an exception was thrown (https://github.com/exercism/clojure/blob/4562cf14857a2b886e27344c2e5e56e20e7f99bc/exercises/wordy/test/wordy_test.clj#L47-L55).

**Comment by Average-user** [comment]
ok thanks.

File: exercises/collatz-conjecture/src/example.clj (hunk 2 above)
Comment (haus) [reply exercises/collatz-conjecture/src/example.clj]: Looking at the other exercises, loop bindings are either on their own lines per pair, or comma separated if there are few. So if you wanted to add a comma between the two pairs, that would also be fine.

**Comment by haus** [comment]
Thanks again @Average-user !

**Comment by Average-user** [comment]
Should i do more?

**Comment by Average-user** [comment]
I think that the source link of the README.md isn't working on the merge

**Comment by haus** [comment]
If you find some other unimplemented exercises that you want to do, we'd love to have them.

I'll try to see what's up with the README.

**Comment by Average-user** [comment]
I've send another pull request with another exercise. But i have no rush.

**Pull request merged**